packages/dashboard/src/util/transformAlarms.ts

```diff
@@ -0,0 +1,29 @@
+import { AlarmQuery } from '@iot-app-kit/source-iotsitewise';
+import {
+  AlarmCompositeModelRequest,
+  AlarmData,
+} from '@iot-app-kit/react-components';
+
+const convertAlarmQueryToAlarmRequest = (alarmQuery: AlarmQuery) => {
+  return (
+    alarmQuery.alarmComponents?.reduce((results, alarmComponent) => {
+      results.push({
+        assetId: alarmQuery.assetId,
+        assetCompositeModelId: alarmComponent.assetCompositeModelId,
+      });
+      return results;
+    }, [] as AlarmCompositeModelRequest[]) ?? []
+  );
+};
+
+export const mapAlarmQueriesToRequests = (alarmQueries: AlarmQuery[]) =>
+  alarmQueries.flatMap((query) => convertAlarmQueryToAlarmRequest(query));
+
+export const transformAlarmForPropertyPanel = (alarm: AlarmData) => ({
+  propertyId: alarm.compositeModelId,
```

Comment:
This isn't exactly the propertyId, how is `propertyId` used to display the property?

Reply:
this is more of an internal naming issue, this array of transformed alarms is a fancy way to figure out what text we want to show users!

Reply:
we basically search through this array of transformed alarms to find one with an ID which matches the "property" we're rendering